### PR TITLE
[BAHIR-155] TTL to set (created setex) and hset

### PR DESCRIPTION
**Pull request by tutss**
This PR introduces the possibility to add a TTL to HSET and SET.
With this new functionality, we're able to expire specific key.
`hset` call executes `jedis.hset`, and if TTL is given, calls `jedis.expire`.

### CODE REVIEW

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java [hunk 1]
```diff
@@ -46,22 +46,57 @@
...
     /**
-     * Use this constructor when data type is {@link RedisDataType#HASH} or {@link RedisDataType#SORTED_SET}.
+     * This additional key is optional for the group {@link RedisDataType#HASH}.
+     * In this case, we need four variables.
+     * <p>For {@link RedisDataType#HASH} we need hash name, hash key, element and TTL.
+     */
+    private Integer additionalTTL;
+
+    /**
+     * Use this constructor when data type is {@link RedisDataType#HASH} with a additional TTL.
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java]: This is incorrect right? You should use this for `SETEX` too?

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java [hunk 2]
```diff
@@ -46,22 +46,57 @@
...
         Objects.requireNonNull(redisCommand, "Redis command type can not be null");
         this.redisCommand = redisCommand;
         this.additionalKey = additionalKey;
+        this.additionalTTL = additionalTTL;
 
         if (redisCommand.getRedisDataType() == RedisDataType.HASH ||
             redisCommand.getRedisDataType() == RedisDataType.SORTED_SET) {
             if (additionalKey == null) {
-                throw new IllegalArgumentException("Hash and Sorted Set should have additional key");
+                throw new IllegalArgumentException("Hash should have additional key");
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java]: Why did you change this?

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java [hunk 3]
```diff
@@ -144,6 +155,9 @@ public void invoke(IN input, Context context) throws Exception {
             case SET:
                 this.redisCommandsContainer.set(key, value);
                 break;
+            case SETEX:
+                this.redisCommandsContainer.setex(key, this.additionalTTL, value);
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java]: Why all commands that uses `SETEX` must have the same TTL? Why can't it be variable as `HSET`?

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java [hunk 4]
```diff
@@ -92,6 +92,13 @@
      * {@code additionalKey} used as set name for {@link RedisDataType#SORTED_SET}
      */
     private String additionalKey;
+
+    /**
+     * This additional time to live is optional for {@link RedisDataType#HASH}.
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java]: and for `SETEX`?

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainer.java [hunk 5]
```diff
@@ -88,6 +91,8 @@
      */
     void set(String key, String value);
 
+    void setex(String key, Integer ttl, String value);
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/container/RedisCommandsContainer.java]: Missing documentation.

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommand.java [hunk 6]
```diff
@@ -45,6 +45,8 @@
      */
     SET(RedisDataType.STRING),
 
+    SETEX(RedisDataType.STRING),
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommand.java]: Missing documentation

File: flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java [hunk 7]
```diff
@@ -22,7 +22,7 @@
 /**
  * The description of the command type. This must be passed while creating new {@link RedisMapper}.
  * <p>When creating descriptor for the group of {@link RedisDataType#HASH} and {@link RedisDataType#SORTED_SET},
- * you need to use first constructor {@link #RedisCommandDescription(RedisCommand, String)}.
```
Comment (gustavo-momente) [inline flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/common/mapper/RedisCommandDescription.java]: Why SORTED_SET Should use this constructor?

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java [hunk 8]
```diff
@@ -239,4 +294,52 @@ public String getValueFromData(Tuple2<String, String> data) {
             return data.f1;
         }
     }
+
+    public static class RedisAdditionalDataAAAMapper implements RedisMapper<Tuple2<String, String>> {
```
Comment (gustavo-momente) [inline flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: Can we change this name?

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java [hunk 9]
```diff
@@ -81,6 +84,22 @@ public void testRedisSetDataType() throws Exception {
         jedis.del(REDIS_KEY);
     }
 
+    @Test
+    public void testRedisStringDataTypeWithTTL() throws Exception {
```
Comment (gustavo-momente) [inline flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: Don't you think we should add the command to the test name?

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java [hunk 10]
```diff
@@ -81,6 +84,22 @@ public void testRedisSetDataType() throws Exception {
...
+                new RedisAdditionalDataAAAMapper(RedisCommand.SETEX));
+
+        source.addSink(redisSink);
+        env.execute("Test Redis Set Data Type With TTL");
+
+        assertEquals(TEST_MESSAGE_LENGTH, jedis.strlen(REDIS_KEY));
+        Thread.sleep(REDIS_TTL * 1000);
+        assertEquals(false, jedis.exists(REDIS_KEY));
+
+        jedis.del(REDIS_ADDITIONAL_KEY);
```
Comment (gustavo-momente) [inline flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: Why are you deleting this ?

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java [hunk 11]
```diff
@@ -127,6 +146,24 @@ public void testRedisHashDataType() throws Exception {
         env.execute("Test Redis Hash Data Type");
 
         assertEquals(NUM_ELEMENTS, jedis.hlen(REDIS_ADDITIONAL_KEY));
+        Thread.sleep(REDIS_TTL * 1000);
+        assertEquals(true, jedis.exists(REDIS_ADDITIONAL_KEY));
```
Comment (gustavo-momente) [inline flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: You can use `jedis.ttl` to check that a key has an associated ttl instead of sleeping

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java [hunk 12]
```diff
@@ -127,6 +146,24 @@ public void testRedisHashDataType() throws Exception {
...
+
+        jedis.del(REDIS_ADDITIONAL_KEY);
+    }
+
+    @Test
+    public void testRedisHashDataTypeWithTTL() throws Exception {
```
Comment (gustavo-momente) [inline flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: You're not testing the case where the TTL comes from the value (the mapper extracts it)

File: flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java (hunk 9 above)
Comment (tutss) [reply flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: Following this test class pattern, every test is specific for each type. I thought this was awkward, but I don't know if 'mixing' what is being tested is a good option. What do you think?

Comment (gustavo-momente) [reply flink-connector-redis/src/test/java/org/apache/flink/streaming/connectors/redis/RedisSinkITCase.java]: But all their tests have the redis command in the name check the previous one it's testing Set and it has Set on the name
